tests/www/dashboard/tests.py

```diff
@@ -1876,138 +1876,330 @@ def test_cannot_switch_to_inactive_company_after_grace_period(self):
+        ]
 
-        recipient.refresh_from_db()
-        assert recipient.notifications
+        # prewarm ContentType cache if needed to avoid extra query
+        ContentType.objects.get_for_model(Company)
 
-        for pk in job_descriptions_pks:
-            assert NewQualifiedJobAppEmployersNotification.is_subscribed(recipient=recipient, subscribed_pk=pk)
+        # No notification settings defined by default
+        assert not NotificationSettings.objects.count()
+        assert not DisabledNotification.objects.count()
 
-    def test_employer_opt_out_company_no_job_descriptions(self):
-        company = CompanyFactory(with_membership=True)
-        user = company.members.first()
-        recipient = user.companymembership_set.get(company=company)
-        form_name = "new_job_app_notification_form"
-        self.client.force_login(user)
+        with self.assertNumQueries(
+            1  # Load django session
+            + 1  # Load current user
+            + 1  # Load company membership
+            + 1  # Load company
+            + 2  # Savepoint and release
+            + 1  # Load employer company
+            + 1  # Load employer notification settings (form init)
+            + 1  # Load employer notification settings (form save)
+            + 3  # Savepoint, create notification settings and release (form save)
+            + 1  # Load notification records (form save)
+            + 3  # Savepoint, update session and release
+        ):
+            response = self.client.post(
+                url, data={notification.__name__: "on" for notification in available_notifications}
+            )
+        assert response.status_code == 302
+        assert response["Location"] == "/dashboard/"
+        assert NotificationSettings.objects.count() == 1
+        assert not DisabledNotification.objects.count()
 
-        # Recipient's notifications are empty for the moment.
-        assert not recipient.notifications
+        # Update, disable all notifications
+        with self.assertNumQueries(
+            1  # Load django session
+            + 1  # Load current user
+            + 1  # Load company membership
+            + 1  # Load company
+            + 2  # Savepoint and release
+            + 1  # Load employer company
+            + 1  # Load employer notification settings (form init)
+            + 1  # Load employer disabled notification (form init)
+            + 1  # Load employer notification settings (form save)
+            + len(available_notifications)  # Load disabled notification record (form save)
+            + 1  # Load notification records (form save)
+            + 1  # Load disabled notifications' notification records (form save)
+            + 1  # Bulk insert disabled notifications (form save)
+        ):
+            response = self.client.post(url, data={"csrfmiddlewaretoken": "NORMALIZED_CSRF_TOKEN"})  # Force POST
```

Comment:
Suggestion valable pour les tests suivants aussi.
```suggestion
            # Send data to bind to the form, otherwise is_valid() returns False
            response = self.client.post(url, {"foo": "bar"})
```